src/main/java/io/jenkins/infra/repository_permissions_updater/hosting/MavenVerifier.java

```diff
@@ -312,40 +319,52 @@ private void checkSoftwareConfigurationManagementField(Model model, HashSet<Veri
-                    " See [here](https://www.jenkins.io/doc/developer/plugin-development/dependency-management/#jenkins-plugin-bom) for details."));
-        }
-        if (hasPluginDependency && bom.isPresent()) {
-            model.getDependencies().stream().filter(d -> d.getGroupId().startsWith("org.jenkins-ci") ||
-                    d.getGroupId().startsWith("io.jenkins") ||
-                    d.getGroupId().equals("org.jenkinsci.plugins") ||
-                    d.getGroupId().equals("org.jvnet.hudson.plugins")).forEach(d -> {
-                if (d.getVersion() != null) {
-                    hostingIssues.add(new VerificationMessage(VerificationMessage.Severity.INFO, "Please check if the dependency `%s:%s` is covered by the bom. If yes then the version should be removed.", d.getGroupId(), d.getArtifactId()));
-                }
-            });
-        }
         JenkinsVersion jenkinsVersion = getJenkinsVersion(model);
-        if (bom.isPresent() && jenkinsVersion != null) {
-            Dependency b = bom.get();
-            String artifactId = b.getArtifactId();
-            artifactId  = artifactId.replace("${jenkins.baseline}", jenkinsVersion.baseline());
-            String expected = "bom-" + jenkinsVersion.baseline() + ".x";
-            if (!expected.equals(artifactId)) {
-                hostingIssues.add(new VerificationMessage(VerificationMessage.Severity.REQUIRED, "The artifactId `%s` of the bom is not in sync with the jenkins baseline version `%s`", artifactId, jenkinsVersion.baseline()));
+        if (jenkinsVersion != null) {
+            HashSet<VerificationMessage> dependencyManagementIssues = new HashSet<>();
+            DependencyManagement depManagement = model.getDependencyManagement();
+            Optional<Dependency> bom = Optional.empty();
+            if (depManagement != null) {
+                bom = depManagement.getDependencies().stream().filter(d ->
+                        d.getGroupId().equals("io.jenkins.tools.bom") &&
+                                d.getArtifactId().startsWith("bom-")
+                ).findFirst();
+            }
+            Set<String> managedDependencies;
+            String bomArtifactId = "bom-" + jenkinsVersion.baseline() + ".x";
+            String latestReleasedBom = getLatestBomVersion(bomArtifactId);
+            if (latestReleasedBom  != null) {
+                managedDependencies = getManagedDependenciesFromBom(bomArtifactId, latestReleasedBom);
+            } else {
+                managedDependencies = Collections.emptySet();
+            }
+            List<Dependency> pluginDependencies = model.getDependencies().stream().filter(d -> managedDependencies.contains(getDependencyAsString(d, false))).toList();
+            if (!pluginDependencies.isEmpty() && bom.isEmpty()) {
```

Comment:
how will this be triggered? as bom is empty, plugin dependencies will never match the above filter?

Reply:
The managedDependencies are collected also when no dependencyManagement section is present or it is there but doesn't include the bom. It is purely based on the jenkins version. See lines 334-340